### ISSUE TITLE
x/tools/gopls: slow completions from github.com/aws/aws-sdk-go/service/ec2

### ISSUE THREAD

**Issue by heschik** [golang/go#37450]
Completing out of the ec2 package is horribly slow, apparently because `astutil.PathEnclosingInterval` doesn't scale. We could add indexes, or stop returning results after the first few hundred.

![profile001](https://user-images.githubusercontent.com/24611692/75279572-cf8f0d80-57d9-11ea-971a-d165e6ddfc5d.png)

@muirdm @stamblerre

**Comment by bmurphy1976** [comment]
Awesome!  I was just coming here to report the same thing (although with much less detail).  I thought my VSCode intellisense was broken, but after debugging it I came to find it was simply taking 12 seconds to load the data for the ec2.EC2 type:

```
[Trace - 17:04:53.416 PM] Received response 'textDocument/completion - (16)' in 12041ms.
```

**Comment by muirdm** [comment]
The proof-of-concept CL above helps a lot, but it is still too slow if you have completion documentation enabled. We can probably speed that up too using the same cache-types.Object-to-ast.Node approach, but I want to hear what others think.

**Comment by mlieberman85** [comment]
Is there anything that can be done from a gopls config level while https://golang.org/cl/221021 is being worked on for gopls to ignore problematic packages like ec2?

**Comment by stamblerre** [comment]
I don't think that will be possible, sorry about that, @mlieberman85. We'll aim to get the fix out in `gopls/v0.5.0`, and I think that https://github.com/golang/go/issues/38278 will be the ultimate fix for this issue.
